src/core/ViewModels/WalletAssetViewModel.hpp

```diff
@@ -29,6 +37,7 @@ public slots:
     void initTransactionsListModel();
 
 private:
+    std::shared_ptr<WalletDataSource> _walletDataSource;
```

Comment:
I would say change this to QPointer, here we need to know only if object is still valid, no need to hold a reference for it.